### PR TITLE
Enqueue custom analytics event to GA

### PR DESCRIPTION
**Pull request by MinhPhu0304**

### CODE REVIEW

File: src/booking/BookingsModal.tsx [hunk 1]
```diff
@@ -166,6 +172,7 @@ const BookingsModal: FunctionComponent<BookingsModalProps> = ({
                             },
                           },
                         }}
+                        onClick={() => enqueueAnalyticsEvent('Click make a booking')}
```
Comment (oeed) [inline src/booking/BookingsModal.tsx:175]: Can we include maybe information about the booking? i.e. the date, far in the future the booking is, what location etc.

File: src/walk-in/modal/WalkInModal.tsx [hunk 2]
```diff
@@ -75,6 +76,7 @@ const WalkInModal: FunctionComponent<Props> = ({
                 },
               }}
               kind={KIND.primary}
+              onClick={() => enqueueAnalyticsEvent('Click get walk in location')}
```
Comment (oeed) [inline src/walk-in/modal/WalkInModal.tsx:79]: Likewise, can we collect a bit more info here?

**Review by oeed — CHANGES_REQUESTED** [review]

File: src/booking/BookingsModal.tsx (hunk 1 above)
Comment (MinhPhu0304) [reply src/booking/BookingsModal.tsx]: done

File: src/RadiusSelect.tsx [hunk 3]
```diff
@@ -36,6 +37,7 @@ export default function RadiusSelect(props: Props) {
           const id = selectedOption.id;
           if (id && typeof id === "number") {
             props.setValue(id);
+            enqueueAnalyticsEvent('Change radius', { radius: id });
```
Comment (noway) [inline src/RadiusSelect.tsx:40]: "Radius changed"

File: src/ShareButtons.tsx [hunk 4]
```diff
@@ -14,32 +14,34 @@ import {
...
+
 export const ShareButtons = () => {
   const shareUrl = "https://vaxx.nz";
   const title = "Vaxx.nz | NZ COVID Vaccination Finder";
 
   return (
     <>
       <div>
-        <FacebookShareButton url={shareUrl} quote={title}>
+        <FacebookShareButton url={shareUrl} quote={title} onClick={() => enqueueAnalyticsEvent('Share by Facebook clicked')}>
```
Comment (noway) [inline src/ShareButtons.tsx:26]: "Share with Facebook clicked"? i think

File: src/booking/BookingsModal.tsx [hunk 5]
```diff
@@ -98,7 +102,10 @@ const BookingsModal: FunctionComponent<BookingsModalProps> = ({
             </ol>
 
             <Button
-              onClick={() => setActiveDate(null)}
+              onClick={() => {
+                enqueueAnalyticsEvent('Click back to Calendar');
```
Comment (noway) [inline src/booking/BookingsModal.tsx:106]: "Back to Calendar clicked"

File: src/booking/BookingsModal.tsx [hunk 6]
```diff
@@ -143,6 +150,7 @@ const BookingsModal: FunctionComponent<BookingsModalProps> = ({
                       href={`https://www.google.com/maps/dir/?api=1&destination=${locationSlotsPair.location.location.lat},${locationSlotsPair.location.location.lng}`}
                       target="_blank"
                       rel="noopener noreferrer"
+                      onClick={() => enqueueAnalyticsEvent('Get direction')}
```
Comment (noway) [inline src/booking/BookingsModal.tsx:153]: "Get Directions clicked"

**Review by oeed — APPROVED** [review]
Looks good to me

File: src/booking/BookingsModal.tsx [hunk 7]
```diff
@@ -166,6 +174,18 @@ const BookingsModal: FunctionComponent<BookingsModalProps> = ({
                             },
                           },
                         }}
+                        onClick={() =>
+                          enqueueAnalyticsEvent('Click make a booking', {
```
Comment (noway) [inline src/booking/BookingsModal.tsx:178]: "Make a Booking clicked"

File: src/booking/BookingsModal.tsx [hunk 8]
```diff
@@ -166,6 +174,18 @@ const BookingsModal: FunctionComponent<BookingsModalProps> = ({
...
+                            location: locationSlotsPair.location,
+                            radiusKm,
+                            spotAvailable: locationSlotsPair.slots?.length || 0,
+                            bookingDateInDays: differenceInDays(parse(
```
Comment (noway) [inline src/booking/BookingsModal.tsx:182]: love it, good shit

**Comment by oeed** [comment]
Have we verified if the analytics are flowing through to the dashboard correctly?

File: src/booking/BookingsModal.tsx [hunk 9]
```diff
@@ -166,6 +174,18 @@ const BookingsModal: FunctionComponent<BookingsModalProps> = ({
                             },
                           },
                         }}
+                        onClick={() =>
+                          enqueueAnalyticsEvent('Click make a booking', {
+                            location: locationSlotsPair.location,
+                            radiusKm,
+                            spotAvailable: locationSlotsPair.slots?.length || 0,
```
Comment (noway) [inline src/booking/BookingsModal.tsx:181]: "spotsAvailable"

File: src/booking/BookingsModal.tsx [hunk 10]
```diff
@@ -207,7 +227,10 @@ const BookingsModal: FunctionComponent<BookingsModalProps> = ({
                 Try changing your search!
               </h1>
               <Button
-                onClick={() => setActiveDate(null)}
+                onClick={() => {
+                  enqueueAnalyticsEvent('Click back to Calendar');
```
Comment (noway) [inline src/booking/BookingsModal.tsx:231]: Back to Calendar clicked

File: src/location-picker/LocationModal.tsx [hunk 11]
```diff
@@ -154,7 +156,10 @@ const LocationModal = (props: Props) => {
           border: "none",
           backgroundColor: "white'",
         }}
-        onClick={() => getLocation()}
+        onClick={() => {
+          enqueueAnalyticsEvent('Use current location click');
```
Comment (noway) [inline src/location-picker/LocationModal.tsx:160]: "Use current location clicked"

File: src/location-picker/LocationPicker.tsx [hunk 12]
```diff
@@ -77,7 +78,10 @@ export const LocationPicker: FunctionComponent<LocationPickerProps> = ({
         <div>
           <Button
             kind={KIND.primary}
-            onClick={() => setIsOpen(true)}
+            onClick={() => {
+              enqueueAnalyticsEvent('Open location modal')
```
Comment (noway) [inline src/location-picker/LocationPicker.tsx:82]: "location modal opened"

**Comment by MinhPhu0304** [comment]
> Have we verified if the analytics are flowing through to the dashboard correctly?

I don't have any access to the dashboard so 🤷

File: src/walk-in/modal/WalkInModal.tsx [hunk 13]
```diff
@@ -75,6 +76,7 @@ const WalkInModal: FunctionComponent<Props> = ({
                 },
               }}
               kind={KIND.primary}
+              onClick={() => enqueueAnalyticsEvent('Click get walk in location', { location })}
```
Comment (noway) [inline src/walk-in/modal/WalkInModal.tsx:79]: "Healthpoint Get Directions clicked"

**Review by noway — CHANGES_REQUESTED** [review]

**Comment by oeed** [comment]
@noway can you verify that the events are showing in the dash

**Comment by MinhPhu0304** [comment]
There's a lag between when we send the data and when we see it in the dashboard. I'm 100% knows if the events actually flows through or not

**Comment by noway** [comment]
> There's a lag between when we send the data and when we see it in the dashboard. I'm 100% knows if the events actually flows through or not

that's fine, that's always the case

**Comment by oeed** [comment]
Okay let's merge if you're happy @noway

**Pull request merged**